src/modules/party-external-rating/party-external-rating.controller.test.ts

```diff
@@ -0,0 +1,105 @@
+    const externalRatingNote2A = valueGenerator.string();
+    const externalRatingUserCode1A = valueGenerator.string();
+    const externalRatingUserCode2A = valueGenerator.string();
+
+    const ratingEntityCodeB = valueGenerator.stringOfNumericCharacters();
+    const assignedRatingCodeB = valueGenerator.stringOfNumericCharacters();
+    const ratedDateB = valueGenerator.date();
+    const probabilityofDefaultB = valueGenerator.probabilityFloat();
+    const lossGivenDefaultB = valueGenerator.nonnegativeFloat();
+    const riskWeightingB = valueGenerator.nonnegativeFloat();
+    const externalRatingNote1B = valueGenerator.string();
+    const externalRatingNote2B = valueGenerator.string();
+    const externalRatingUserCode1B = valueGenerator.string();
+    const externalRatingUserCode2B = valueGenerator.string();
+
+    const externalRatings = [
+      {
+        partyIdentifier: partyIdentifier,
+        ratingEntity: {
+          ratingEntityCode: ratingEntityCodeA,
+        },
+        assignedRating: {
+          assignedRatingCode: assignedRatingCodeA,
+        },
+        ratedDate: ratedDateA,
+        probabilityofDefault: probabilityofDefaultA,
+        lossGivenDefault: lossGivenDefaultA,
+        riskWeighting: riskWeightingA,
+        externalRatingNote1: externalRatingNote1A,
+        externalRatingNote2: externalRatingNote2A,
+        externalRatingUserCode1: externalRatingUserCode1A,
+        externalRatingUserCode2: externalRatingUserCode2A,
+      },
+      {
+        partyIdentifier: partyIdentifier,
+        ratingEntity: {
+          ratingEntityCode: ratingEntityCodeB,
+        },
+        assignedRating: {
+          assignedRatingCode: assignedRatingCodeB,
+        },
+        ratedDate: ratedDateB,
+        probabilityofDefault: probabilityofDefaultB,
+        lossGivenDefault: lossGivenDefaultB,
+        riskWeighting: riskWeightingB,
+        externalRatingNote1: externalRatingNote1B,
+        externalRatingNote2: externalRatingNote2B,
+        externalRatingUserCode1: externalRatingUserCode1B,
+        externalRatingUserCode2: externalRatingUserCode2B,
+      },
+    ];
+
+    let partyExternalRatingsProvider: PartyExternalRatingsProvider;
+    let controller: PartyExternalRatingController;
+
+    beforeEach(() => {
+      partyExternalRatingsProvider = {
+        getExternalRatingsForParty: jest.fn(),
+      };
+      controller = new PartyExternalRatingController(partyExternalRatingsProvider);
+    });
+
+    it('returns the external ratings for the party from the provider', async () => {
+      // eslint-disable-next-line jest/unbound-method
+      when(partyExternalRatingsProvider.getExternalRatingsForParty).calledWith(partyIdentifier).mockResolvedValueOnce(externalRatings);
+
+      const ratings = await controller.getExternalRatingsForParty(partyIdentifier);
+
+      expect(ratings).toStrictEqual(externalRatings);
+    });
+
+    it('does not return unexpected keys from the response', async () => {
```

Comment:
can we put `not` all in uppercase? so we know it's a special scenario

Reply:
Sure - do we want to capitalise `not` in all test case names?

Reply:
yes please. as that's a special scenario and we can easily identify them